universal-application-tool-0.0.1/app/repository/ProgramRepository.java

```diff
@@ -170,6 +171,35 @@ private BlockDefinition updateQuestionVersions(BlockDefinition block, Transactio
         executionContext);
   }
 
+  public Program updateOrCreateDraft(Program existingProgram) throws ProgramNotFoundException {
+    Optional<Program> existingDraft =
+        ebeanServer
+            .find(Program.class)
+            .where()
+            .eq("lifecycle_stage", LifecycleStage.DRAFT.getValue())
+            .eq("name", existingProgram.getProgramDefinition().name())
+            .findOneOrEmpty();
+    if (existingDraft.isPresent()) {
+      Program updatedDraft =
+          existingProgram.getProgramDefinition().toBuilder()
+              .setId(existingDraft.get().id)
+              .setLifecycleStage(LifecycleStage.DRAFT)
+              .build()
+              .toProgram();
+      this.updateProgramSync(updatedDraft);
+      return updatedDraft;
```

Comment:
Suggestion (non-blocking): can we return the existing draft here directly?

Reply:
We could return the existing draft without updates since right now it's kind of a bug that lets you create two drafts of the same program, but if we ever wanted that we'd want it to update the existing draft.

Reply:
SG, feel free to ignore.